sweep/sweeper.go

```diff
@@ -111,6 +111,28 @@ type inputCluster struct {
 	inputs       pendingInputs
 }
 
+// pendingSweepsReq is an internal message we'll use to represent an external
+// caller's intent to retrieve all of the pending inputs the UtxoSweeper is
+// attempting to sweep.
+type pendingSweepsReq struct {
+	respChan chan []PendingSweep
+}
+
+// PendingInput represents an input the UtxoSweeper is currently attempting to
+// sweep.
+type PendingInput struct {
+	OutPoint wire.OutPoint
+	FeeRate  lnwallet.SatPerKWeight
+}
+
+// PendingSweep represents a set of inputs that are grouped together according
+// to some heuristic in order to sweep them within a single transaction with a
+// specific sweep fee rate.
+type PendingSweep struct {
+	FeeRate lnwallet.SatPerKWeight
```

Comment:
I am not sure what the sweep fee rate is actually going to tell you. If a tx were to be published now, it would be published with that fee rate. But it could very well be that we already published a tx with this input in it at a different fee rate within the same bucket range. The sweep fee rate fluctuates because it is an average that may change.

Reply:
Removed. I think it would still be useful to show which inputs are grouped together though, no? They're currently sorted in ascending fee rate order.

Reply:
I think we could do that, but not sure if you should really group the inputs. There are also other ways of grouping, like broadcast attempts, next height, etc. Why group in this way? It isn't even guaranteed that those inputs actually end up in the same tx.

You could also add two fields `bucketMin` and `bucketMax` to show the bucket it is in. Or a separate list in the response that contains the static bucket boundaries.

I would keep it simple and flat initially and see how it is used.